heron/api/src/java/com/twitter/heron/api/HeronSubmitter.java

```diff
@@ -91,33 +97,33 @@ public static void submitTopology(String name, Config heronConfig, HeronTopology
     TopologyUtils.validateTopology(fTopology);
     assert fTopology.isInitialized();
 
-    if (heronCmdOptions.get("cmdline.topologydefn.tmpdirectory") != null) {
-      submitTopologyToFile(fTopology, heronCmdOptions);
-    } else {
-      throw new RuntimeException("topology definition temp directory not specified");
-    }
+    submitTopologyToFile(fTopology, heronCmdOptions);
   }
 
-  // Submits to the file
+  /**
+   * Submits a topology to definition file
+   *
+   * @param fTopology the processing to execute.
+   * @param heronCmdOptions the commandline options.
+   * @throws TopologySubmissionException if the topology submission is failed
+   */
   private static void submitTopologyToFile(TopologyAPI.Topology fTopology,
                                            Map<String, String> heronCmdOptions) {
-    String dirName = heronCmdOptions.get("cmdline.topologydefn.tmpdirectory");
+    String dirName = heronCmdOptions.get(CMDLINE_TOPOLOGY_DEFINITION_TMP_DIRECTORY);
     if (dirName == null || dirName.isEmpty()) {
-      throw new RuntimeException("Improper specification of directory");
+      throw new TopologySubmissionException("Topology definition temp directory not specified. "
+          + "Please set cmdline option: " + CMDLINE_TOPOLOGY_DEFINITION_TMP_DIRECTORY);
     }
-    String fileName = dirName + "/" + fTopology.getName() + ".defn";
-    BufferedOutputStream bos = null;
-    try {
-      //create an object of FileOutputStream
-      FileOutputStream fos = new FileOutputStream(new File(fileName));
-      //create an object of BufferedOutputStream
-      bos = new BufferedOutputStream(fos);
+
+    String fileName = dirName + "/" + fTopology.getName() + TOPOLOGY_DEFINITION_SUFFIX;
```

Comment:
@erenavsarogullari - instead of forward slash - can you use Java Path class - to keep it more generic?

Reply:
Sure, will be addressing.